src/diart/models.py

```diff
@@ -113,8 +114,17 @@ def duration(self) -> float:
         self.load()
         return self.model.specifications.duration
 
-    def forward(self, waveform: torch.Tensor) -> torch.Tensor:
-        return self.model(waveform)
+    def __call__(self, waveform: torch.Tensor) -> torch.Tensor:
+        """
+        Call the forward pass of the segmentation model.
+        Parameters
+        ----------
+        waveform: torch.Tensor, shape (batch, channels, samples)
+        Returns
+        -------
+        speaker_segmentation: torch.Tensor, shape (batch, frames, speakers)
+        """
+        return super().__call__(waveform)
 
 
 class EmbeddingModel(LazyModel):
```

Comment:
Add a `__call__` method to this class:

```python
    def __call__(
        self, waveform: torch.Tensor, weights: Optional[torch.Tensor] = None
    ) -> torch.Tensor:
        return super().__call__(waveform, weights)
```

This simply works as a marker. Making sure that users creating their own `EmbeddingModel` know what arguments to expect here

Reply:
if I'm adding a `__call__` to `EmbeddingModel`, it would be reasonable to also add one to `SegmentationModel`. I will add the methods in question but for now flag them as `abstractmethod`. If you would prefer the same implementation as in the `Pyannote...` models, I can add them but I considered it easier to subclass this way.

Reply:
Sidenote: if you want me to move the whole `__call__` to `SegmentationModel` (see [here](https://github.com/juanmc2005/diart/pull/188#discussion_r1383821730)), I will do the same for EmbeddingModel to keep consistency.